### DISCUSSION TITLE
Install alpine plugin

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#5357] [Help]
How can I install alpine plugin in filament? If it was in a basic laravel project I would do it, but in Filament, I have no idea where to do it. I want to install this plugin: [https://alpinejs.dev/plugins/intersect](https://alpinejs.dev/plugins/intersect)
I put this code in `resources/js/app.js` but no result:
```js
import Alpine from 'alpinejs'
import intersect from '@alpinejs/intersect'

Alpine.plugin(intersect)

window.Alpine = Alpine

Alpine.start()
```
```bash
npm run dev
```

![image](https://user-images.githubusercontent.com/29732308/209864249-131eca8c-cd73-4af0-8ff1-797568fd839c.png)

although I don't even know if this is right or not.

**Comment by danharrin** [comment]
Register it with `alpine:init`? Since we install Alpine for you

**Reply by ArtMin96** [reply]
I was just trying this. I know that alpine has been installed in filament and has its own app.js, the problem is how can I add a plugin?

**Reply by danharrin** [reply]
Follow the instructions as if Alpine was loaded via CDN

https://alpinejs.dev/plugins/intersect

**Reply by ArtMin96** [reply]
I wouldn't want to use CDN, but that option doesn't work either. I added this to the service provider but no change, the same console error
```php
Filament::registerScripts([
    'https://unpkg.com/@alpinejs/intersect@3.x.x/dist/cdn.min.js',
]);
```

**Comment by danharrin** [comment]
Try

```php
Filament::registerScripts([
    'https://unpkg.com/@alpinejs/intersect@3.x.x/dist/cdn.min.js',
], shouldBeLoadedBeforeCoreScripts: true);
```

**Reply by ArtMin96** [reply]
The console warning was cleared, but the problem was not resolved. If I add `shouldBeLoadedBeforeCoreScripts: true` to `Filament::registerScripts`, the alpine plugin (intersect) will load earlier than alpine js.

![user-menu](https://user-images.githubusercontent.com/29732308/210156859-9c7d91d7-554b-46ca-b8e5-c43232f13752.png)

**Reply by ArtMin96** [reply]
![user-menu](https://user-images.githubusercontent.com/29732308/210168031-9b0a6cfe-b2a8-4320-9f4f-1273447046ef.png)

**Reply by ArtMin96** [reply]
Imo, it is impossible to add a plugin at the moment because you have to register the plugin before `Alpine.start()`.

In app.js

```js
import Alpine from 'alpinejs'
// ...
import Tooltip from '@ryangjchandler/alpine-tooltip'
import intersect from '@alpinejs/intersect' // <- Push this

// ...
Alpine.plugin(Tooltip)
Alpine.plugin(intersect) // <- Push this

// ...

Alpine.start()
```

**Reply by C4NH4M** [reply]
Although this wasn't the problem I was facing, @ArtMin96 had helped me by pointing out where to add my plugin code as on the official docs all it says is:

You can install Collapse from NPM for use inside your bundle like so:

npm install @alpinejs/collapse

> _Then initialize it from your bundle:_

import Alpine from 'alpinejs'
import collapse from '@alpinejs/collapse'
 
Alpine.plugin(collapse)
...

At no point does it explain where this is. In my case It was indeed in **app.js**

**Comment by a-coding-a** [comment]
It seems like you cannot? install an alpinejs plugin via NPM, instead we need to load it via the CDN script.
